darwin/dataset/remote_dataset.py

```diff
@@ -78,61 +83,74 @@ def __init__(
 
         Parameters
         ----------
-        files_to_upload : list[Path]
-            List of files to upload. It can be a folder.
+        files_to_upload : Optional[List[Union[str, Path, LocalFile]]]
+            List of files to upload. Those can be folders.
         blocking : bool
-            If False, the dataset is not uploaded and a generator function is returned instead
+            If False, the dataset is not uploaded and a generator function is returned instead.
         multi_threaded : bool
             Uses multiprocessing to upload the dataset in parallel.
             If blocking is False this has no effect.
-        files_to_exclude : list[str]
-            List of files to exclude from the file scan (which is done only if files is None)
+        files_to_exclude : Optional[Union[str, Path]]]
+            Optional list of files to exclude from the file scan. Those can be folders.
         fps : int
-            Number of file per seconds to upload
+            When the uploading file is a video, specify its framerate.
         as_frames: bool
-            Annotate as video.
-        path: str
-            Optional path to put the files into
-        progress_callback: (total_file_count, file_advancement, Optional[file_name], file_total_bytes, file_bytes_sent) => None
-            Optional callback, called with the total number of files to upload, how far the current file has uploaded and how many bytes have been read.
+            When the uploading file is a video, specify whether it's going to be uploaded as a list of frames.
+        path: Optional[str]
+            Optional path to store the files in.
+        progress_callback: Optional[ProgressCallback]
+            Optional callback, called every time the progress of an uploading files is reported.
+        file_upload_callback: Optional[FileUploadCallback]
+            Optional callback, called every time a file chunk is uploaded.
         Returns
         -------
         handler : UploadHandler
            Class for handling uploads, progress and error messages
         """
 
-        # Init optional parameters
         if files_to_exclude is None:
             files_to_exclude = []
+
         if files_to_upload is None:
             raise ValueError("No files or directory specified.")
 
-        files_to_upload = find_files(files=files_to_upload, recursive=True, files_to_exclude=files_to_exclude)
+        generic_parameters_specified = path is not None or fps != 0 or as_frames
+        if any(isinstance(item, LocalFile) for item in files_to_upload) and generic_parameters_specified:
+            raise ValueError("Cannot specify a path when uploading a LocalFile object.")
 
-        if not files_to_upload:
-            raise ValueError("No files to upload, check your path, exclusion filters and resume flag")
+        uploading_files = [item for item in files_to_upload if isinstance(item, LocalFile)]
```

Comment:
would this be correct?
```
uploading_files: LocalFile = [item for item in files_to_upload if isinstance(item, LocalFile)]
```

Reply:
I think so! What's your opinion about explicit type annotations on variables? Your linter should be able to figure it out anyway, and I have a feeling that being stricter would cause a lot of code to be unnecessarily split into multiple lines, thus making it a bit less readable

Reply:
I personally like static typing. For me it was not obvious that `uploading_files` was a list of `LocalFile`. I had to spend time analyzing the code and then try to make an educated guess of what the type was. This could have been avoided if the type was explicit. 

I dont mind the extra boiler plate, which imo, is actually useful documentation and not really useless boiler plate. The linter is also not that smart most times, so it needs a hand. 

I personally prefer having things with types. If this means more lines, that's ok, that's why we use code formatters and good practices to make the code readable.